Source/Core/Common/SFMLHelper.cpp

```diff
@@ -0,0 +1,36 @@
+// Copyright 2018 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#include "Common/SFMLHelper.h"
+
+#include "Common/Swap.h"
+
+namespace Common
+{
+u64 PacketReadU64(sf::Packet& packet)
+{
+  u32 low, high;
+  packet >> low >> high;
+  return low | (static_cast<u64>(high) << 32);
+}
+
+void PacketWriteU64(sf::Packet& packet, const u64 value)
+{
+  packet << static_cast<u32>(value) << static_cast<u32>(value >> 32);
+}
+
+void PacketReadBigEndian(sf::Packet& packet, BigEndianValue<u16>& bev)
```

Comment:
This kind of defeats the purpose of returning BigEndianValue at all. The only reason this function exists is because we can't use the normal `>>` operator on `sf::Packet` with BigEndianValue.

Reply:
How so? This simply makes:

```cpp
Common::PacketReadBigEndian(packet, header.banner_size);
```

become:

```cpp
header.banner_size = Common::PacketReadBigEndian(packet);
```

Reply:
Yes, and then the return type can simply be changed to `u16` as the `=` operator of `BigEndianValue` handles that. So basically the function just turns into `PacketReadU16` instead of doing any BigEndianValue handling, which I guess isn't bad, but seems a bit redundant as `sf::Packet` itself handles `u16` using the `>>` and `<<` operators.

Reply:
> Yes, and then the return type can be changed to simply u16 as the = operator of BigEndianValue handles that. So basically the function just turns into PacketReadU16 instead of doing any BigEndianValue handling,

The one argument constructor of BigEndianValue performs the same behavior as its `operator=`, so they fundamentally perform the exact same behavior. I don't get what you mean about it becoming essentially like PacketReadU16.

Reply:
I just mean that having it return a `BigEndianValue` at all is redundant.

Reply:
Oh OK